src/plugins/auto/src/plugin.cpp

```diff
@@ -623,22 +635,31 @@ std::list<DeviceInformation> Plugin::get_valid_device(
     if (is_default_list) {
         // Generate the default device priority for selecting logic of AUTO.
         // Default priority of selecting device: dGPU > iGPU > 3rd part devices > CPU
-        valid_devices.clear();
-        valid_devices.splice(valid_devices.end(), dGPU);
-        valid_devices.splice(valid_devices.end(), iGPU);
-        valid_devices.splice(valid_devices.end(), Others);
-        valid_devices.splice(valid_devices.end(), CPU);
-        return valid_devices;
+        std::list<DeviceInformation> default_valid_devices;
+        default_valid_devices.splice(default_valid_devices.end(), dGPU);
+        default_valid_devices.splice(default_valid_devices.end(), iGPU);
+        default_valid_devices.splice(default_valid_devices.end(), Others);
+        default_valid_devices.splice(default_valid_devices.end(), CPU);
+        return default_valid_devices.empty() ? (valid_filtered_devices.empty() ? valid_devices : valid_filtered_devices)
+                                             : default_valid_devices;
     }
     // sort validDevices
     valid_devices.sort([](const DeviceInformation& a, const DeviceInformation& b) {
         return a.device_priority < b.device_priority;
     });
-    return valid_devices;
+    if (valid_filtered_devices.empty())
+        return valid_devices;
+
+    valid_filtered_devices.sort([](const DeviceInformation& a, const DeviceInformation& b) {
+        return a.device_priority < b.device_priority;
```

Comment:
Define common lambda to compare device priority and use instead repeat it?

Reply:
Done.